service/src/test/java/bio/terra/pipelines/service/PipelinesServiceTest.java

```diff
@@ -170,4 +189,212 @@ void updatePipelineWorkspaceId() {
+
+    String[] stringArrayWithNull = new String[] {null, "has a null value"};
+
+    return Stream.of(
+        // arguments are: type specification, value to be tested, isRequired, expected errorMessage
+        // basic type checks
+        arguments("Integer", "I am a string", true, commonTypeErrorMessage),
+        arguments("Integer", 2.3, true, commonTypeErrorMessage),
+        arguments("Integer", "2.3", true, commonTypeErrorMessage),
+        arguments("String", "", true, notEmptyErrorMessage),
+        arguments("String", "", false, notEmptyErrorMessage),
+        arguments(
+            "String", List.of("this", "is", "not", "a", "string"), true, commonTypeErrorMessage),
+        arguments("VCF", "path/to/file.vcf", true, vcfTypeErrorMessage),
+        arguments("VCF", 3, true, vcfTypeErrorMessage),
+        arguments(
+            "Array_VCF",
+            new String[] {"path/to/file.vcf.gz", "not a path"},
+            true,
+            arrayVCFTypeErrorMessage),
+        arguments("Array_String", "I am a string", true, commonTypeErrorMessage),
+        arguments(
+            "Array_String",
+            List.of("", "there's an empty string"),
+            true,
+            arrayStringTypeErrorMessage),
+        arguments(
+            "Array_String",
+            new String[] {null, "list with null"},
+            true,
+            arrayStringTypeErrorMessage),
+        arguments("Integer", "I am a string", false, commonTypeErrorMessage),
+        // null and empty checks
+        arguments("String", null, false, notNullErrorMessage),
+        arguments("Integer", null, true, notNullErrorMessage),
+        arguments("Array_String", new String[] {}, true, emptyArrayErrorMessage),
+        arguments("Array_VCF", new String[] {}, true, emptyArrayErrorMessage),
+        arguments("Array_String", null, true, notNullErrorMessage),
+        arguments("Array_String", stringArrayWithNull, true, arrayStringTypeErrorMessage),
+        arguments("Array_VCF", null, true, notNullErrorMessage),
+        arguments(
+            "Array_VCF",
+            new String[] {null, "list/with/null.vcf.gz"},
+            true,
+            arrayVCFTypeErrorMessage));
+  }
+
+  @ParameterizedTest
+  @MethodSource("inputTypeValidationsNotValid")
+  void validateInputTypeNotValid(
+      String type, Object inputValue, Boolean isRequired, String errorMessage) {
+    PipelineInputDefinition inputDefinition =
+        new PipelineInputDefinition(1L, "input_name", type, isRequired);
+    List<PipelineInputDefinition> inputDefinitions = new ArrayList<>(List.of(inputDefinition));
+
+    LinkedHashMap<String, Object> inputs = new LinkedHashMap<>();
+    inputs.put("input_name", inputValue);
+
+    assertEquals(
+        errorMessage.formatted(type),
+        pipelinesService.validateInputTypes(inputDefinitions, inputs).get(0));
+  }
+
+  @Test
+  void validateOptionalInputTypeValidAbsent() {
```

Comment:
you could modify this to a parameterized test where you test the different configuration of optional/required and present/absent.